Failing Update Check prevents Cryptomater v1.5.5 to show main window

### Description

When launching Cryptomater v1.5.5 nothing appears on the desktop.  Taskbar icon is visible but cannot open preferences from here either.  Only option that works is "quit".  

This was the case on a newly reinstalled desktop running windows 10.  In the end, I found the old installer v1.4.15 which reinstalled Dokan and Cryptomater and it launched properly.  After upgrading back to v1.5.5, same issue happened.  

### System Setup

Windows Version:	10.0.18363 Build 18363
Cryptomator version:  1.5.5
Drive: WebDAV (I believe because it is the only thing that I saw in the installation settings)  

### Steps to Reproduce

1. Install v1.5.5
2. Launch Cryptomater

#### Expected Behavior

I expect to see the application window.  

#### Actual Behavior

No application window appears.  

#### Reproducibility

Always

### Additional Information

cryptomator0.log
`17:44:30.901 [main] INFO  org.cryptomator.launcher.Cryptomator - Starting Cryptomator 1.5.5 on Windows 10 10.0 (amd64)`
`17:44:31.552 [JavaFX Application Thread] INFO  o.c.ui.launcher.FxApplicationStarter - JavaFX Runtime started.`
`17:44:31.563 [JavaFX Application Thread] INFO  org.cryptomator.jni.FunctionsLoader - loaded WinFunctions.dll`
`17:44:39.730 [main] INFO  org.cryptomator.launcher.Cryptomator - UI shut down`

Note, I cannot launch app to enable debug mode.  

[cryptomator0.log](https://github.com/cryptomator/cryptomator/files/4774381/cryptomator0.log)

Thanks for the feedback.  

Following the steps, I first got an error that it could not load WinFunctions.dll.  After, I added the Cryptomator path to the user path and then launched again.  

Next, I got the following errors.  

`21:22:23.322 [JavaFX Application Thread] INFO  org.cryptomator.jni.FunctionsLoader - loaded WinFunctions.dll
 vsync: true vpipe: true
Exception in thread "JavaFX Application Thread" java.lang.InternalError: java.io.IOException: Unable to establish loopback connection
        at java.net.http/jdk.internal.net.http.HttpClientImpl.<init>(HttpClientImpl.java:313)
        at java.net.http/jdk.internal.net.http.HttpClientImpl.create(HttpClientImpl.java:255)
        at java.net.http/jdk.internal.net.http.HttpClientBuilderImpl.build(HttpClientBuilderImpl.java:135)
        at java.net.http/java.net.http.HttpClient.newHttpClient(HttpClient.java:158)
        at org.cryptomator.ui.fxapp.UpdateCheckerModule.provideHttpClient(UpdateCheckerModule.java:39)
        at org.cryptomator.ui.fxapp.UpdateCheckerModule_ProvideHttpClientFactory.provideHttpClient(UpdateCheckerModule_ProvideHttpClientFactory.java:27)
        at org.cryptomator.ui.fxapp.UpdateCheckerModule_ProvideHttpClientFactory.get(UpdateCheckerModule_ProvideHttpClientFactory.java:19)
        at org.cryptomator.ui.fxapp.UpdateCheckerModule_ProvideHttpClientFactory.get(UpdateCheckerModule_ProvideHttpClientFactory.java:8)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at org.cryptomator.ui.fxapp.UpdateCheckerModule_ProvideCheckForUpdatesServiceFactory.get(UpdateCheckerModule_ProvideCheckForUpdatesServiceFactory.java:43)
        at org.cryptomator.ui.fxapp.UpdateCheckerModule_ProvideCheckForUpdatesServiceFactory.get(UpdateCheckerModule_ProvideCheckForUpdatesServiceFactory.java:14)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at org.cryptomator.ui.fxapp.UpdateChecker_Factory.get(UpdateChecker_Factory.java:45)
        at org.cryptomator.ui.fxapp.UpdateChecker_Factory.get(UpdateChecker_Factory.java:12)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at org.cryptomator.ui.mainwindow.MainWindowTitleController_Factory.get(MainWindowTitleController_Factory.java:47)
        at org.cryptomator.ui.mainwindow.MainWindowTitleController_Factory.get(MainWindowTitleController_Factory.java:12)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at org.cryptomator.ui.common.FXMLLoaderFactory.constructController(FXMLLoaderFactory.java:76)
        at javafx.fxml.FXMLLoader$ValueElement.processAttribute(FXMLLoader.java:938)
        at javafx.fxml.FXMLLoader$InstanceDeclarationElement.processAttribute(FXMLLoader.java:980)
        at javafx.fxml.FXMLLoader$Element.processStartElement(FXMLLoader.java:227)
        at javafx.fxml.FXMLLoader$ValueElement.processStartElement(FXMLLoader.java:752)
        at javafx.fxml.FXMLLoader.processStartElement(FXMLLoader.java:2722)
        at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2552)
        at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2466)
        at javafx.fxml.FXMLLoader$IncludeElement.constructValue(FXMLLoader.java:1154)
        at javafx.fxml.FXMLLoader$ValueElement.processStartElement(FXMLLoader.java:754)
        at javafx.fxml.FXMLLoader.processStartElement(FXMLLoader.java:2722)
        at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2552)
        at javafx.fxml.FXMLLoader.load(FXMLLoader.java:2450)
        at org.cryptomator.ui.common.FXMLLoaderFactory.load(FXMLLoaderFactory.java:47)
        at org.cryptomator.ui.common.FXMLLoaderFactory.createScene(FXMLLoaderFactory.java:61)
        at org.cryptomator.ui.mainwindow.MainWindowModule.provideMainScene(MainWindowModule.java:67)
        at org.cryptomator.ui.mainwindow.MainWindowModule_ProvideMainSceneFactory.provideMainScene(MainWindowModule_ProvideMainSceneFactory.java:36)
        at org.cryptomator.ui.mainwindow.MainWindowModule_ProvideMainSceneFactory.get(MainWindowModule_ProvideMainSceneFactory.java:27)
        at org.cryptomator.ui.mainwindow.MainWindowModule_ProvideMainSceneFactory.get(MainWindowModule_ProvideMainSceneFactory.java:10)
        at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at org.cryptomator.ui.mainwindow.MainWindowComponent.showMainWindow(MainWindowComponent.java:27)
        at org.cryptomator.ui.fxapp.FxApplication.lambda$showMainWindow$1(FxApplication.java:94)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
        at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.io.IOException: Unable to establish loopback connection
        at java.base/sun.nio.ch.PipeImpl$Initializer.run(PipeImpl.java:94)
        at java.base/sun.nio.ch.PipeImpl$Initializer.run(PipeImpl.java:61)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:554)
        at java.base/sun.nio.ch.PipeImpl.<init>(PipeImpl.java:171)
        at java.base/sun.nio.ch.SelectorProviderImpl.openPipe(SelectorProviderImpl.java:50)
        at java.base/java.nio.channels.Pipe.open(Pipe.java:155)
        at java.base/sun.nio.ch.WindowsSelectorImpl.<init>(WindowsSelectorImpl.java:142)
        at java.base/sun.nio.ch.WindowsSelectorProvider.openSelector(WindowsSelectorProvider.java:44)
        at java.base/java.nio.channels.Selector.open(Selector.java:295)
        at java.net.http/jdk.internal.net.http.HttpClientImpl$SelectorManager.<init>(HttpClientImpl.java:710)
        at java.net.http/jdk.internal.net.http.HttpClientImpl.<init>(HttpClientImpl.java:310)
        ... 46 more
Caused by: java.net.ConnectException: Connection refused: connect
        at java.base/sun.nio.ch.Net.connect0(Native Method)
        at java.base/sun.nio.ch.Net.connect(Net.java:503)
        at java.base/sun.nio.ch.Net.connect(Net.java:492)
        at java.base/sun.nio.ch.SocketChannelImpl.connect(SocketChannelImpl.java:751)
        at java.base/java.nio.channels.SocketChannel.open(SocketChannel.java:194)
        at java.base/sun.nio.ch.PipeImpl$Initializer$LoopbackConnector.run(PipeImpl.java:127)
        at java.base/sun.nio.ch.PipeImpl$Initializer.run(PipeImpl.java:76)
        ... 56 more
`

Seeming, there is a loopback connection issue that is particular to my PC.  I realized there is one way I use my PC that is unique to most others is that I run VPN full time.  After disabling VPN, it worked with the method explained above as well as launching the app normally.  

I guess we can say cause of the issue is now clear.  But now it brings up the question, why can't I use Cryptomator while VPN is running?

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -12,7 +12,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-java@v1
         with:
-          java-version: 14
+          java-version: 16
       - uses: actions/cache@v1
         with:
           path: ~/.m2/repository

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -28,10 +28,10 @@ jobs:
         if: github.repository == 'cryptomator/cryptomator'
         run: |
           curl -o ~/codacy-coverage-reporter.jar https://repo.maven.apache.org/maven2/com/codacy/codacy-coverage-reporter/7.1.0/codacy-coverage-reporter-7.1.0-assembly.jar
-          $JAVA_HOME/bin/java -jar ~/codacy-coverage-reporter.jar report -l Java -r main/commons/target/site/jacoco/jacoco.xml --partial
-          $JAVA_HOME/bin/java -jar ~/codacy-coverage-reporter.jar report -l Java -r main/ui/target/site/jacoco/jacoco.xml --partial
-          $JAVA_HOME/bin/java -jar ~/codacy-coverage-reporter.jar report -l Java -r main/launcher/target/site/jacoco/jacoco.xml --partial
-          $JAVA_HOME/bin/java -jar ~/codacy-coverage-reporter.jar final
+          $JAVA_HOME/bin/java --illegal-access=permit -jar ~/codacy-coverage-reporter.jar report -l Java -r main/commons/target/site/jacoco/jacoco.xml --partial
+          $JAVA_HOME/bin/java --illegal-access=permit -jar ~/codacy-coverage-reporter.jar report -l Java -r main/ui/target/site/jacoco/jacoco.xml --partial
+          $JAVA_HOME/bin/java --illegal-access=permit -jar ~/codacy-coverage-reporter.jar report -l Java -r main/launcher/target/site/jacoco/jacoco.xml --partial
+          $JAVA_HOME/bin/java --illegal-access=permit -jar ~/codacy-coverage-reporter.jar final
         env:
           CODACY_PROJECT_TOKEN: ${{ secrets.CODACY_PROJECT_TOKEN }}
       - name: Assemble buildkit-linux.zip

--- a/.idea/misc.xml
+++ b/.idea/misc.xml
@@ -8,7 +8,7 @@
       </list>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_14" default="false" project-jdk-name="14" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_16" project-jdk-name="16" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/out" />
   </component>
 </project>

--- a/README.md
+++ b/README.md
@@ -65,7 +65,7 @@ For more information on the security details visit [cryptomator.org](https://doc
 
 ### Dependencies
 
-* JDK 14 (e.g. adoptopenjdk)
+* JDK 16 (e.g. adoptopenjdk)
 * Maven 3
 * Optional: OS-dependent build tools for native packaging (see [Windows](https://github.com/cryptomator/cryptomator-win), [OS X](https://github.com/cryptomator/cryptomator-osx), [Linux](https://github.com/cryptomator/builder-containers))
 

--- a/main/commons/src/main/java/org/cryptomator/common/CommonsModule.java
+++ b/main/commons/src/main/java/org/cryptomator/common/CommonsModule.java
@@ -47,10 +47,12 @@ public abstract class CommonsModule {
 	@Named("licensePublicKey")
 	static String provideLicensePublicKey() {
 		// in PEM format without the dash-escaped begin/end lines
-		return "MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQB7NfnqiZbg2KTmoflmZ71PbXru7oW" //
-				+ "fmnV2yv3eDjlDfGruBrqz9TtXBZV/eYWt31xu1osIqaT12lKBvZ511aaAkIBeOEV" //
-				+ "gwcBIlJr6kUw7NKzeJt7r2rrsOyQoOG2nWc/Of/NBqA3mIZRHk5Aq1YupFdD26QE" //
-				+ "r0DzRyj4ixPIt38CQB8=";
+		return """
+				MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQB7NfnqiZbg2KTmoflmZ71PbXru7oW\
+				fmnV2yv3eDjlDfGruBrqz9TtXBZV/eYWt31xu1osIqaT12lKBvZ511aaAkIBeOEV\
+				gwcBIlJr6kUw7NKzeJt7r2rrsOyQoOG2nWc/Of/NBqA3mIZRHk5Aq1YupFdD26QE\
+				r0DzRyj4ixPIt38CQB8=\
+				""";
 	}
 
 	@Provides

--- a/main/commons/src/main/java/org/cryptomator/common/LicenseChecker.java
+++ b/main/commons/src/main/java/org/cryptomator/common/LicenseChecker.java
@@ -33,8 +33,8 @@ class LicenseChecker {
 		try {
 			byte[] keyBytes = BaseEncoding.base64().decode(pemEncodedPublicKey);
 			PublicKey key = KeyFactory.getInstance("EC").generatePublic(new X509EncodedKeySpec(keyBytes));
-			if (key instanceof ECPublicKey) {
-				return (ECPublicKey) key;
+			if (key instanceof ECPublicKey k) {
+				return k;
 			} else {
 				throw new IllegalStateException("Key not an EC public key.");
 			}

--- a/main/commons/src/main/java/org/cryptomator/common/settings/VaultSettings.java
+++ b/main/commons/src/main/java/org/cryptomator/common/settings/VaultSettings.java
@@ -173,8 +173,7 @@ public class VaultSettings {
 
 	@Override
 	public boolean equals(Object obj) {
-		if (obj instanceof VaultSettings && obj.getClass().equals(this.getClass())) {
-			VaultSettings other = (VaultSettings) obj;
+		if (obj instanceof VaultSettings other && obj.getClass().equals(this.getClass())) {
 			return Objects.equals(this.id, other.id);
 		} else {
 			return false;

--- a/main/commons/src/main/java/org/cryptomator/common/vaults/Vault.java
+++ b/main/commons/src/main/java/org/cryptomator/common/vaults/Vault.java
@@ -355,8 +355,7 @@ public class Vault {
 
 	@Override
 	public boolean equals(Object obj) {
-		if (obj instanceof Vault && obj.getClass().equals(this.getClass())) {
-			final Vault other = (Vault) obj;
+		if (obj instanceof Vault other && obj.getClass().equals(this.getClass())) {
 			return Objects.equals(this.vaultSettings, other.vaultSettings);
 		} else {
 			return false;

--- a/main/commons/src/main/java/org/cryptomator/common/vaults/VaultListChangeListener.java
+++ b/main/commons/src/main/java/org/cryptomator/common/vaults/VaultListChangeListener.java
@@ -22,10 +22,10 @@ class VaultListChangeListener implements ListChangeListener<Vault> {
 	public void onChanged(Change<? extends Vault> c) {
 		while (c.next()) {
 			if (c.wasAdded()) {
-				List<VaultSettings> addedSettings = c.getAddedSubList().stream().map(Vault::getVaultSettings).collect(Collectors.toList());
+				List<VaultSettings> addedSettings = c.getAddedSubList().stream().map(Vault::getVaultSettings).toList();
 				vaultSettingsList.addAll(c.getFrom(), addedSettings);
 			} else if (c.wasRemoved()) {
-				List<VaultSettings> removedSettings = c.getRemoved().stream().map(Vault::getVaultSettings).collect(Collectors.toList());
+				List<VaultSettings> removedSettings = c.getRemoved().stream().map(Vault::getVaultSettings).toList();
 				vaultSettingsList.removeAll(removedSettings);
 			}
 		}

--- a/main/commons/src/main/java/org/cryptomator/common/vaults/VaultListManager.java
+++ b/main/commons/src/main/java/org/cryptomator/common/vaults/VaultListManager.java
@@ -79,7 +79,7 @@ public class VaultListManager {
 	}
 
 	private void addAll(Collection<VaultSettings> vaultSettings) {
-		Collection<Vault> vaults = vaultSettings.stream().map(this::create).collect(Collectors.toList());
+		Collection<Vault> vaults = vaultSettings.stream().map(this::create).toList();
 		vaultList.addAll(vaults);
 	}
 

--- a/main/commons/src/test/java/org/cryptomator/common/EnvironmentTest.java
+++ b/main/commons/src/test/java/org/cryptomator/common/EnvironmentTest.java
@@ -32,7 +32,7 @@ class EnvironmentTest {
 	public void testSettingsPath() {
 		System.setProperty("cryptomator.settingsPath", "~/.config/Cryptomator/settings.json:~/.Cryptomator/settings.json");
 
-		List<Path> result = env.getSettingsPath().collect(Collectors.toList());
+		List<Path> result = env.getSettingsPath().toList();
 		MatcherAssert.assertThat(result, Matchers.hasSize(2));
 		MatcherAssert.assertThat(result, Matchers.contains(Paths.get("/home/testuser/.config/Cryptomator/settings.json"), //
 				Paths.get("/home/testuser/.Cryptomator/settings.json")));
@@ -43,7 +43,7 @@ class EnvironmentTest {
 	public void testIpcPortPath() {
 		System.setProperty("cryptomator.ipcPortPath", "~/.config/Cryptomator/ipcPort.bin:~/.Cryptomator/ipcPort.bin");
 
-		List<Path> result = env.getIpcPortPath().collect(Collectors.toList());
+		List<Path> result = env.getIpcPortPath().toList();
 		MatcherAssert.assertThat(result, Matchers.hasSize(2));
 		MatcherAssert.assertThat(result, Matchers.contains(Paths.get("/home/testuser/.config/Cryptomator/ipcPort.bin"), //
 				Paths.get("/home/testuser/.Cryptomator/ipcPort.bin")));
@@ -54,7 +54,7 @@ class EnvironmentTest {
 	public void testKeychainPath() {
 		System.setProperty("cryptomator.keychainPath", "~/AppData/Roaming/Cryptomator/keychain.json");
 
-		List<Path> result = env.getKeychainPath().collect(Collectors.toList());
+		List<Path> result = env.getKeychainPath().toList();
 		MatcherAssert.assertThat(result, Matchers.hasSize(1));
 		MatcherAssert.assertThat(result, Matchers.contains(Paths.get("/home/testuser/AppData/Roaming/Cryptomator/keychain.json")));
 	}
@@ -88,7 +88,7 @@ class EnvironmentTest {
 		@DisplayName("test.path.property=")
 		public void testEmptyList() {
 			System.setProperty("test.path.property", "");
-			List<Path> result = env.getPaths("test.path.property").collect(Collectors.toList());
+			List<Path> result = env.getPaths("test.path.property").toList();
 
 			MatcherAssert.assertThat(result, Matchers.hasSize(0));
 		}
@@ -97,7 +97,7 @@ class EnvironmentTest {
 		@DisplayName("test.path.property=/foo/bar/test")
 		public void testSingleAbsolutePath() {
 			System.setProperty("test.path.property", "/foo/bar/test");
-			List<Path> result = env.getPaths("test.path.property").collect(Collectors.toList());
+			List<Path> result = env.getPaths("test.path.property").toList();
 
 			MatcherAssert.assertThat(result, Matchers.hasSize(1));
 			MatcherAssert.assertThat(result, Matchers.hasItem(Paths.get("/foo/bar/test")));
@@ -107,7 +107,7 @@ class EnvironmentTest {
 		@DisplayName("test.path.property=~/test")
 		public void testSingleHomeRelativePath() {
 			System.setProperty("test.path.property", "~/test");
-			List<Path> result = env.getPaths("test.path.property").collect(Collectors.toList());
+			List<Path> result = env.getPaths("test.path.property").toList();
 
 			MatcherAssert.assertThat(result, Matchers.hasSize(1));
 			MatcherAssert.assertThat(result, Matchers.hasItem(Paths.get("/home/testuser/test")));
@@ -117,7 +117,7 @@ class EnvironmentTest {
 		@DisplayName("test.path.property=~/test:~/test2:/foo/bar/test")
 		public void testMultiplePaths() {
 			System.setProperty("test.path.property", "~/test:~/test2:/foo/bar/test");
-			List<Path> result = env.getPaths("test.path.property").collect(Collectors.toList());
+			List<Path> result = env.getPaths("test.path.property").toList();
 
 			MatcherAssert.assertThat(result, Matchers.hasSize(3));
 			MatcherAssert.assertThat(result, Matchers.contains(Paths.get("/home/testuser/test"), //

--- a/main/commons/src/test/java/org/cryptomator/common/LicenseCheckerTest.java
+++ b/main/commons/src/test/java/org/cryptomator/common/LicenseCheckerTest.java
@@ -9,10 +9,12 @@ import java.util.Optional;
 
 class LicenseCheckerTest {
 
-	private static final String PUBLIC_KEY = "MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQBgc4HZz+/fBbC7lmEww0AO3NK9wVZ" //
-			+ "PDZ0VEnsaUFLEYpTzb90nITtJUcPUbvOsdZIZ1Q8fnbquAYgxXL5UgHMoywAib47" //
-			+ "6MkyyYgPk0BXZq3mq4zImTRNuaU9slj9TVJ3ScT3L1bXwVuPJDzpr5GOFpaj+WwM" //
-			+ "Al8G7CqwoJOsW7Kddns=";
+	private static final String PUBLIC_KEY = """
+			MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQBgc4HZz+/fBbC7lmEww0AO3NK9wVZ\
+			PDZ0VEnsaUFLEYpTzb90nITtJUcPUbvOsdZIZ1Q8fnbquAYgxXL5UgHMoywAib47\
+			6MkyyYgPk0BXZq3mq4zImTRNuaU9slj9TVJ3ScT3L1bXwVuPJDzpr5GOFpaj+WwM\
+			Al8G7CqwoJOsW7Kddns=\
+			""";
 
 	private LicenseChecker licenseChecker;
 

--- a/main/commons/src/test/java/org/cryptomator/common/settings/SettingsJsonAdapterTest.java
+++ b/main/commons/src/test/java/org/cryptomator/common/settings/SettingsJsonAdapterTest.java
@@ -21,13 +21,18 @@ public class SettingsJsonAdapterTest {
 
 	@Test
 	public void testDeserialize() throws IOException {
-		String vault1Json = "{\"id\": \"1\", \"path\": \"/vault1\", \"mountName\": \"vault1\", \"winDriveLetter\": \"X\"}";
-		String vault2Json = "{\"id\": \"2\", \"path\": \"/vault2\", \"mountName\": \"vault2\", \"winDriveLetter\": \"Y\"}";
-		String json = "{\"directories\": [" + vault1Json + "," + vault2Json + "]," //
-				+ "\"checkForUpdatesEnabled\": true,"//
-				+ "\"port\": 8080,"//
-				+ "\"numTrayNotifications\": 42,"//
-				+ "\"preferredVolumeImpl\": \"FUSE\"}";
+		String json = """
+				{
+					"directories": [
+						{"id": "1", "path": "/vault1", "mountName": "vault1", "winDriveLetter": "X"},
+						{"id": "2", "path": "/vault2", "mountName": "vault2", "winDriveLetter": "Y"}
+					],
+					"checkForUpdatesEnabled": true,
+					"port": 8080,
+					"numTrayNotifications": 42,
+					"preferredVolumeImpl": "FUSE"
+				}
+				""";
 
 		Settings settings = adapter.fromJson(json);
 

--- a/main/launcher/src/main/java/org/cryptomator/launcher/FileOpenRequestHandler.java
+++ b/main/launcher/src/main/java/org/cryptomator/launcher/FileOpenRequestHandler.java
@@ -41,7 +41,7 @@ class FileOpenRequestHandler {
 	}
 
 	private void openFiles(OpenFilesEvent evt) {
-		Collection<Path> pathsToOpen = evt.getFiles().stream().map(File::toPath).collect(Collectors.toList());
+		Collection<Path> pathsToOpen = evt.getFiles().stream().map(File::toPath).toList();
 		AppLaunchEvent launchEvent = new AppLaunchEvent(AppLaunchEvent.EventType.OPEN_FILE, pathsToOpen);
 		tryToEnqueueFileOpenRequest(launchEvent);
 	}
@@ -59,7 +59,7 @@ class FileOpenRequestHandler {
 				LOG.trace("Argument not a valid path: {}", str);
 				return null;
 			}
-		}).filter(Objects::nonNull).collect(Collectors.toList());
+		}).filter(Objects::nonNull).toList();
 		if (!pathsToOpen.isEmpty()) {
 			AppLaunchEvent launchEvent = new AppLaunchEvent(AppLaunchEvent.EventType.OPEN_FILE, pathsToOpen);
 			tryToEnqueueFileOpenRequest(launchEvent);

--- a/main/launcher/src/main/java/org/cryptomator/logging/LoggerModule.java
+++ b/main/launcher/src/main/java/org/cryptomator/logging/LoggerModule.java
@@ -47,8 +47,8 @@ public class LoggerModule {
 	@Singleton
 	static LoggerContext provideLoggerContext() {
 		ILoggerFactory loggerFactory = LoggerFactory.getILoggerFactory();
-		if (loggerFactory instanceof LoggerContext) {
-			return (LoggerContext) loggerFactory;
+		if (loggerFactory instanceof LoggerContext context) {
+			return context;
 		} else {
 			throw new IllegalStateException("SLF4J not bound to Logback.");
 		}

--- a/main/pom.xml
+++ b/main/pom.xml
@@ -22,7 +22,7 @@
 
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-		<project.jdk.version>14</project.jdk.version>
+		<project.jdk.version>16</project.jdk.version>
 
 		<!-- cryptomator dependencies -->
 		<cryptomator.cryptofs.version>1.9.14</cryptomator.cryptofs.version>

--- a/main/ui/src/main/java/org/cryptomator/ui/common/DefaultSceneFactory.java
+++ b/main/ui/src/main/java/org/cryptomator/ui/common/DefaultSceneFactory.java
@@ -42,8 +42,8 @@ public class DefaultSceneFactory implements Function<Parent, Scene> {
 	protected void configureScene(Scene scene) {
 		scene.windowProperty().addListener(observable -> {
 			Window window = scene.getWindow();
-			if (window instanceof Stage) {
-				setupDefaultAccelerators(scene, (Stage) window);
+			if (window instanceof Stage s) {
+				setupDefaultAccelerators(scene, s);
 			}
 		});
 	}

--- a/main/ui/src/main/java/org/cryptomator/ui/unlock/UnlockWorkflow.java
+++ b/main/ui/src/main/java/org/cryptomator/ui/unlock/UnlockWorkflow.java
@@ -76,8 +76,8 @@ public class UnlockWorkflow extends Task<Boolean> {
 
 		setOnFailed(event -> {
 			Throwable throwable = event.getSource().getException();
-			if (throwable instanceof InvalidMountPointException) {
-				handleInvalidMountPoint((InvalidMountPointException) throwable);
+			if (throwable instanceof InvalidMountPointException e) {
+				handleInvalidMountPoint(e);
 			} else {
 				handleGenericError(throwable);
 			}

--- a/main/ui/src/test/java/org/cryptomator/ui/recoverykey/RecoveryKeyFactoryTest.java
+++ b/main/ui/src/test/java/org/cryptomator/ui/recoverykey/RecoveryKeyFactoryTest.java
@@ -48,18 +48,22 @@ class RecoveryKeyFactoryTest {
 	@Test
 	@DisplayName("validateRecoveryKey() with invalid checksum")
 	public void testValidateValidateRecoveryKeyWithInvalidCrc() {
-		boolean result = inTest.validateRecoveryKey("pathway lift abuse plenty export texture gentleman landscape beyond ceiling around leaf cafe" //
-				+ " charity border breakdown victory surely computer cat linger restrict infer crowd live computer true written amazed investor boot" //
-				+ " depth left theory snow whereby terminal weekly reject happiness circuit partial cup wrong");
+		boolean result = inTest.validateRecoveryKey("""
+				pathway lift abuse plenty export texture gentleman landscape beyond ceiling around leaf cafe charity \
+				border breakdown victory surely computer cat linger restrict infer crowd live computer true written amazed \
+				investor boot depth left theory snow whereby terminal weekly reject happiness circuit partial cup wrong \
+				""");
 		Assertions.assertFalse(result);
 	}
 
 	@Test
 	@DisplayName("validateRecoveryKey() with valid input")
 	public void testValidateValidateRecoveryKeyWithValidKey() {
-		boolean result = inTest.validateRecoveryKey("pathway lift abuse plenty export texture gentleman landscape beyond ceiling around leaf cafe" //
-				+ " charity border breakdown victory surely computer cat linger restrict infer crowd live computer true written amazed investor boot" //
-				+ " depth left theory snow whereby terminal weekly reject happiness circuit partial cup ad");
+		boolean result = inTest.validateRecoveryKey("""
+				pathway lift abuse plenty export texture gentleman landscape beyond ceiling around leaf cafe charity \
+				border breakdown victory surely computer cat linger restrict infer crowd live computer true written amazed \
+				investor boot depth left theory snow whereby terminal weekly reject happiness circuit partial cup ad \
+				""");
 		Assertions.assertTrue(result);
 	}
 

